web/src/compose_recipient.ts

```diff
@@ -31,6 +31,11 @@ type MessageType = "stream" | "private";
 let compose_select_recipient_dropdown_widget: DropdownWidget;
 
 function composing_to_current_topic_narrow(): boolean {
+    // If the narrow state's stream name is undefined, then
+    // the user cannot be composing to a current topic narrow.
+    if (narrow_state.stream_name() === undefined) {
```

Comment:
Can we use `stream_id` here? I guess we should also convert the rest of this function's check to just compare the `compose_state.stream_id === narrow_state.stream_id`, so it'd be a second commit. But in general we should get out of the business of using stream name comparisons when we have an ID.

Reply:
Sure thing. I'll get that implemented and also the function check.